src/MakeJsonCommand.php

```diff
@@ -127,14 +137,74 @@ public function __invoke( $args, $assoc_args ) {
 		WP_CLI::success( sprintf( 'Created %d %s.', $result_count, Utils\pluralize( 'file', $result_count ) ) );
 	}
 
+	/**
+	 * Collect maps from paths, normalize and merge
+	 *
+	 * @param string|bool $paths argument. False to do nothing.
+	 * @return array|null Mapping array. Null if no maps specified.
+	 */
+	protected function build_map( $paths ) {
+		if ( false === $paths ) {
+			return null;
+		}
+
+		$map = [];
+
+		if ( true === $paths ) {
+			$paths = [ 'build' . DIRECTORY_SEPARATOR . 'map.json' ];
+		} else {
+			$paths = array_unique( array_filter( explode( ',', $paths ) ) );
+		}
+		WP_CLI::debug( sprintf( 'Using files %s', implode( ', ', $paths ) ), 'make-json' );
+
+		foreach ( $paths as $path ) {
+			if ( ! file_exists( $path ) || is_dir( $path ) ) {
+				WP_CLI::warning( sprintf( 'File %s does not exist', $path ) );
+				continue;
+			}
+
+			$json = json_decode( file_get_contents( $path ), true );
+			if ( ! is_array( $json ) ) {
+				WP_CLI::warning( sprintf( 'File %s contained invalid map', $path ) );
+				continue;
+			}
+
+			$key_num = count( $json );
+			// normalize contents to string[]
+			$json = array_map(
+				function ( $value ) {
+					if ( is_array( $value ) ) {
+						$value = array_values( array_filter( $value, 'is_string' ) );
+						if ( ! empty( $value ) ) {
+							return $value;
+						}
+					}
+
+					if ( is_string( $value ) ) {
+						return [ $value ];
+					}
+
+					return null;
+				},
+				$json
+			);
+			WP_CLI::debug( sprintf( 'Dropped %d keys from %s', count( $json ) - $key_num, $path ), 'make-json' );
+
+			$map = array_merge_recursive( $map, $json );
+		}
+
+		return $map;
+	}
+
 	/**
 	 * Splits a single PO file into multiple JSON files.
 	 *
 	 * @param string $source_file Path to the source file.
 	 * @param string $destination Path to the destination directory.
+	 * @param array  $map               Source to build file mapping.
```

Comment:
```suggestion
	 * @param array|null  $map               Source to build file mapping.
```

According to the code below and `build_map()`, it can also be `null`

Although we could simplify the code below if it were never `null`, but simply an empty `array` instead.

Reply:
But if it were never null, we'd always have to have a map, since translations not in the map are dropped